lib/scheduler.js

```diff
@@ -172,19 +173,31 @@ class Scheduler {
         }
 
         if (schedule.time) {
-            let start = schedule.time.start;
-            let times;
-            if (start === 'sunrise' || start === 'sunset') {
-                times = this.suncalc.getTimes(new Date(), this.latitude, this.longitude);
-                start = times[start];
+            let start = schedule.time.start,
+                end = schedule.time.end,
+                now = new this.Date(),
+                times,
+                astroNameStart = this._getAstroName(start),
+                astroNameEnd = this._getAstroName(end);
+
+            if (now.getDate() !== this.todaysAstroTimes['sunrise']) {
+                this._setAstroVars();
+            }
+            if (astroNameStart) {
+                times = this.todaysAstroTimes;
```

Comment:
is it correct that variable "times" is only used in this block? Then use block scoped var ... so kill the "parent block "times" variable and create a new let times" here and another in the other if block. or is the value used later too? I think not

Reply:
`times` was intentionally declared there because it was like this before :-) :-) But i also thougt that that was correct because it is used at different places as kind of a tempVar.

thinking about, different assignments to it make it worth to get declared when used? Is that what you want to tell me?

Reply:
yes